Content.Client/Administration/AdminNameOverlay.cs

```diff
@@ -1,23 +1,31 @@
+using System.Linq;
 using System.Numerics;
 using Content.Client.Administration.Systems;
+using Content.Shared.CCVar;
+using Content.Shared.Mind;
 using Robust.Client.Graphics;
 using Robust.Client.ResourceManagement;
 using Robust.Client.UserInterface;
-using Robust.Shared;
-using Robust.Shared.Enums;
 using Robust.Shared.Configuration;
+using Robust.Shared.Enums;
+using Robust.Shared.Prototypes;
 
 namespace Content.Client.Administration;
 
 internal sealed class AdminNameOverlay : Overlay
 {
     private readonly AdminSystem _system;
+    private readonly IConfigurationManager _config = IoCManager.Resolve<IConfigurationManager>(); // TODO is this even legal?
     private readonly IEntityManager _entityManager;
     private readonly IEyeManager _eyeManager;
     private readonly EntityLookupSystem _entityLookup;
     private readonly IUserInterfaceManager _userInterfaceManager;
     private readonly Font _font;
 
+    //TODO make this adjustable via GUI
+    private readonly ProtoId<RoleTypePrototype>[] _filter =
+        ["SoloAntagonist", "TeamAntagonist", "SiliconAntagonist", "FreeAgent"];
+
     public AdminNameOverlay(AdminSystem system, IEntityManager entityManager, IEyeManager eyeManager, IResourceCache resourceCache, EntityLookupSystem entityLookup, IUserInterfaceManager userInterfaceManager)
     {
         _system = system;
```

Comment:
```suggestion
    {
        IoCManager.InjectDependencies(this);

        _system = system;
```
I just gave it a test and this works perfectly fine. I guess you forgot to do the dependency injection, which is why it was null.